### ISSUE TITLE
maybe issuer in luvit.

### ISSUE THREAD

**Issue by zhaozg** [luvit/luvit#739]
= 1. checkout luvit and run make to get luvi,lit,luvit on windows 7 64bits successfully.

```ps
E:\work\luaclibs\luvit>make
"Building lit"
Downloading https://github.com/luvit/luvi/releases/download/v2.0.9/luvi-regular-
Windows-amd64.exe to luvi.exe
Downloading https://github.com/luvit/lit/archive/1.2.9.zip to lit.zip
lit version: 1.2.9
luvi version: v2.0.9
command: make lit.zip
load config: D:\zhaozg\AppData\Roaming\litconfig
creating binary: lit.exe
copying binary prefix: 2583040 bytes from E:\work\luaclibs\luvit\luvi.exe
importing: E:\work\luaclibs\luvit\lit.zip
including: commands/README
including: commands/add.lua
including: commands/auth.lua
including: commands/claim.lua
including: commands/config.lua
including: commands/down.lua
including: commands/help.lua
including: commands/install.lua
including: commands/make.lua
including: commands/publish.lua
including: commands/serve.lua
including: commands/share.lua
including: commands/sync.lua
including: commands/unclaim.lua
including: commands/up.lua
including: commands/update.lua
including: init.lua
including: libs/api.lua
including: libs/auto-updater.lua
including: libs/autocore.lua
including: libs/codec.lua
including: libs/core.lua
including: libs/db-fs.lua
including: libs/db.lua
including: libs/exec.lua
including: libs/github-request.lua
including: libs/handlers.lua
including: libs/log.lua
including: libs/pkg.lua
including: libs/rdb.lua
including: libs/rules.lua
including: libs/storage.lua
including: libs/vfs.lua
including: main.lua
including: package.lua
skipping: tests
compiling: commands/add.lua
compiling: commands/auth.lua
compiling: commands/claim.lua
compiling: commands/config.lua
compiling: commands/down.lua
compiling: commands/help.lua
compiling: commands/install.lua
compiling: commands/make.lua
compiling: commands/publish.lua
compiling: commands/serve.lua
compiling: commands/share.lua
compiling: commands/sync.lua
compiling: commands/unclaim.lua
compiling: commands/up.lua
compiling: commands/update.lua
compiling: init.lua
compiling: libs/api.lua
compiling: libs/auto-updater.lua
compiling: libs/autocore.lua
compiling: libs/codec.lua
compiling: libs/core.lua
compiling: libs/db-fs.lua
compiling: libs/db.lua
compiling: libs/exec.lua
compiling: libs/github-request.lua
compiling: libs/handlers.lua
compiling: libs/log.lua
compiling: libs/pkg.lua
compiling: libs/rdb.lua
compiling: libs/rules.lua
compiling: libs/storage.lua
compiling: libs/vfs.lua
compiling: main.lua
compiling: package.lua
adding: creationix/websocket-codec@1.0.2-1 (deps/websocket-codec.lua)
compiling: deps/websocket-codec.lua
adding: creationix/coro-channel@1.1.0 (deps/coro-channel.lua)
compiling: deps/coro-channel.lua
adding: creationix/coro-tcp@1.0.5-2 (deps/coro-tcp.lua)
compiling: deps/coro-tcp.lua
adding: creationix/coro-fs@1.2.3-1 (deps/coro-fs.lua)
compiling: deps/coro-fs.lua
adding: creationix/prompt@1.0.3-1 (deps/prompt.lua)
compiling: deps/prompt.lua
adding: creationix/ssh-rsa@1.0.0-1 (deps/ssh-rsa.lua)
compiling: deps/ssh-rsa.lua
adding: luvit/pretty-print@1.0.2-1 (deps/pretty-print.lua)
compiling: deps/pretty-print.lua
adding: creationix/coro-tls@1.1.4-1 (deps/coro-tls.lua)
compiling: deps/coro-tls.lua
adding: creationix/coro-http@1.0.7-1 (deps/coro-http.lua)
compiling: deps/coro-http.lua
adding: creationix/hex-bin@1.0.0-1 (deps/hex-bin.lua)
compiling: deps/hex-bin.lua
adding: luvit/json@2.5.0 (deps/json.lua)
compiling: deps/json.lua
adding: creationix/semver@1.0.2-1 (deps/semver.lua)
compiling: deps/semver.lua
adding: creationix/coro-wrapper@1.0.0-1 (deps/coro-wrapper.lua)
compiling: deps/coro-wrapper.lua
adding: luvit/readline@1.1.0-1 (deps/readline.lua)
compiling: deps/readline.lua
adding: creationix/git@1.0.1-1 (deps/git.lua)
compiling: deps/git.lua
adding: luvit/require@1.2.0 (deps/require.lua)
compiling: deps/require.lua
adding: luvit/http-codec@1.0.0-1 (deps/http-codec.lua)
compiling: deps/http-codec.lua
done building: lit.exe
done: success

"Building luvit"
lit version: 1.2.9
luvi version: v2.0.9
command: make
load config: D:\zhaozg\AppData\Roaming\litconfig
creating binary: luvit.exe
copying binary prefix: 2583040 bytes from E:\work\luaclibs\luvit\lit.exe
importing: E:\work\luaclibs\luvit
skipping: bench
skipping: examples
including: init.lua
including: main.lua
including: package.lua
skipping: tests
compiling: init.lua
compiling: main.lua
compiling: package.lua
adding: luvit/https@1.0.1-3 (E:\work\luaclibs\luvit\deps\https.lua)
compiling: deps/https.lua
adding: luvit/helpful@1.0.0-1 (E:\work\luaclibs\luvit\deps\helpful.lua)
compiling: deps/helpful.lua
adding: luvit/hooks@1.0.0-1 (E:\work\luaclibs\luvit\deps\hooks.lua)
compiling: deps/hooks.lua
adding: luvit/net@1.1.1-1 (E:\work\luaclibs\luvit\deps\net.lua)
compiling: deps/net.lua
adding: luvit/querystring@1.0.0-1 (E:\work\luaclibs\luvit\deps\querystring.lua)
compiling: deps/querystring.lua
adding: luvit/require@1.2.0 (E:\work\luaclibs\luvit\deps\require.lua)
compiling: deps/require.lua
adding: luvit/codec@1.0.0-1 (E:\work\luaclibs\luvit\deps\codec.lua)
compiling: deps/codec.lua
adding: luvit/http@1.1.2 (E:\work\luaclibs\luvit\deps\http.lua)
compiling: deps/http.lua
adding: luvit/fs@1.2.0 (E:\work\luaclibs\luvit\deps\fs.lua)
compiling: deps/fs.lua
adding: luvit/url@1.0.4-1 (E:\work\luaclibs\luvit\deps\url.lua)
compiling: deps/url.lua
adding: luvit/childprocess@1.0.4-1 (E:\work\luaclibs\luvit\deps\childprocess.lua
)
compiling: deps/childprocess.lua
adding: luvit/dns@1.0.0-3 (E:\work\luaclibs\luvit\deps\dns.lua)
compiling: deps/dns.lua
adding: luvit/pretty-print@1.0.2-1 (E:\work\luaclibs\luvit\deps\pretty-print.lua
)
compiling: deps/pretty-print.lua
adding: luvit/process@1.1.0-2 (E:\work\luaclibs\luvit\deps\process.lua)
compiling: deps/process.lua
adding: luvit/http-codec@1.0.0-1 (E:\work\luaclibs\luvit\deps\http-codec.lua)
compiling: deps/http-codec.lua
adding: luvit/buffer@1.0.1-1 (E:\work\luaclibs\luvit\deps\buffer.lua)
compiling: deps/buffer.lua
adding: luvit/tls@1.2.0 (E:\work\luaclibs\luvit\deps\tls/)
compiling: deps/tls/common.lua
compiling: deps/tls/init.lua
compiling: deps/tls/lcrypto.lua
compiling: deps/tls/root_ca.lua
adding: luvit/timer@1.0.0-1 (E:\work\luaclibs\luvit\deps\timer.lua)
compiling: deps/timer.lua
adding: luvit/json@2.5.0 (E:\work\luaclibs\luvit\deps\json.lua)
compiling: deps/json.lua
adding: luvit/path@1.0.0-1 (E:\work\luaclibs\luvit\deps\path/)
compiling: deps/path/base.lua
compiling: deps/path/init.lua
adding: luvit/repl@1.0.1-1 (E:\work\luaclibs\luvit\deps\repl.lua)
compiling: deps/repl.lua
adding: luvit/utils@1.0.0-1 (E:\work\luaclibs\luvit\deps\utils.lua)
compiling: deps/utils.lua
adding: luvit/core@1.0.2-1 (E:\work\luaclibs\luvit\deps\core.lua)
compiling: deps/core.lua
adding: luvit/readline@1.1.0-1 (E:\work\luaclibs\luvit\deps\readline.lua)
compiling: deps/readline.lua
adding: luvit/los@1.0.0-1 (E:\work\luaclibs\luvit\deps\los.lua)
compiling: deps/los.lua
adding: luvit/dgram@1.1.0-1 (E:\work\luaclibs\luvit\deps\dgram.lua)
compiling: deps/dgram.lua
adding: luvit/stream@1.1.0-1 (E:\work\luaclibs\luvit\deps\stream/)
compiling: deps/stream/init.lua
compiling: deps/stream/stream_core.lua
compiling: deps/stream/stream_duplex.lua
compiling: deps/stream/stream_observable.lua
compiling: deps/stream/stream_passthrough.lua
compiling: deps/stream/stream_readable.lua
compiling: deps/stream/stream_transform.lua
compiling: deps/stream/stream_writable.lua
done building: luvit.exe
done: success
```

= 2. checkout luvit.io to launcher it with luvit
```ps
E:\work\luaapps\luvit\luvit.io>\work\luaclibs\luvit\lit.exe install
lit version: 1.2.9
luvi version: v2.0.9
command: install
load config: D:\zhaozg\AppData\Roaming\litconfig
version mismatch (disk): coro-tls 1.2.0 ~= 1.1.3
done: success


E:\work\luaapps\luvit\luvit.io>\work\luaclibs\luvit\luvit.exe server
Uncaught exception:
[string "E:/work/luaapps/luvit/luvit.io/deps/hoedown/i..."]:11: attempt to conca
tenate a nil value
stack traceback:
        [string "E:/work/luaapps/luvit/luvit.io/deps/hoedown/i..."]:11: in funct
ion 'fn'
        [string "bundle:deps/require.lua"]:275: in function 'require'
        [string "E:/work/luaapps/luvit/luvit.io/libs/markdown.lua"]:1: in functi
on 'fn'
        [string "bundle:deps/require.lua"]:275: in function 'require'
        [string "E:/work/luaapps/luvit/luvit.io/libs/render-te..."]:5: in functi
on 'fn'
        [string "bundle:deps/require.lua"]:275: in function 'require'
        [string "E:/work/luaapps/luvit/luvit.io/libs/controlle..."]:1: in functi
on 'fn'
        [string "bundle:deps/require.lua"]:275: in function 'require'
        [string "E:\work\luaapps\luvit\luvit.io\server.lua"]:3: in function 'fn'

        [string "bundle:deps/require.lua"]:275: in function 'require'
        [string "bundle:main.lua"]:115: in function 'main'
        [string "bundle:init.lua"]:34: in function <[string "bundle:init.lua"]:3
2>
        [C]: in function 'xpcall'
        [string "bundle:init.lua"]:32: in function 'fn'
        [string "bundle:deps/require.lua"]:275: in function <[string "bundle:dep
s/require.lua"]:240>
```
Any advice or bugs?

**Comment by creationix** [comment]
No idea what's the issue, I'll test on my windows box.

**Comment by creationix** [comment]
Yep, I can confirm the bug on windows.  I'll see what the issue is.

**Comment by zhaozg** [comment]
bacause missing hoedown?

**Comment by creationix** [comment]
Yep, hoedown was missing, but even after adding it, I have new issues.

**Comment by creationix** [comment]
Try the `make.bat test` command in https://github.com/creationix/lit-hoedown to see the issue. (You may need to manually include the submodules, not sure)

**Comment by creationix** [comment]
Ok, the issue was that they use the old `.def` file method for exporting symbols.  I have an updated dll published to lit as hoedown@1.1.1 that should work on Windows-x64.

**Comment by creationix** [comment]
I just published lit 2.0.2 that has many windows fixes.  I can now confirm that building and running the sample hoedown app works on windows.

```
> lit make github://creationix/howdown-sample
> render.exe
```
